smart_open/tests/test_smart_open.py

```diff
@@ -168,6 +168,28 @@ def test_webhdfs_read(self):
         smart_open_object = smart_open.WebHdfsOpenRead(smart_open.ParseUri("webhdfs://127.0.0.1:8440/path/file"))
         self.assertEqual(smart_open_object.read().decode("utf-8"), "line1\nline2")
 
+    def test_http_readline(self):
+        """Can we stream from an HTTP site correctly"""
+        with smart_open.HttpOpenRead(smart_open.ParseUri('http://www.google.com'), 'r') as smart_open_object:
+            content = smart_open_object.readline().decode('utf-8')
+            expected_str = '<!doctype html>'
+            self.assertEqual(content[:len(expected_str)], expected_str)
+
+    def test_http_read(self):
+        """Can we perform chunked reads on an HTTP stream correctly"""
+        with smart_open.HttpOpenRead(smart_open.ParseUri('http://www.google.com'), 'r') as smart_open_object:
+            expected_str = '<!doctype html>'
+            content = (smart_open_object.read(3).decode('utf-8') + 
+                       smart_open_object.read(len(expected_str) - 3).decode('utf-8'))
+            self.assertEqual(content[:len(expected_str)], expected_str)
+
+    def test_https_readline(self):
+        """Can we stream from an HTTPS site correctly"""
+        with smart_open.HttpOpenRead(smart_open.ParseUri('https://www.google.com'), 'r') as smart_open_object:
+            content = smart_open_object.readline().decode('utf-8')
+            expected_str = '<!doctype html>'
+            self.assertEqual(content[:len(expected_str)], expected_str)
```

Comment:
OK, fixed. I added a test for Basic http auth, and made the other tests not actually go to google.com. I didn't add a Kerbeos test because I'm not sure that's easily doable without running within a Kerborized environment.